pkg/operator/mon/mon.go

```diff
@@ -306,25 +328,79 @@ func (c *Cluster) createService(mon *monConfig) (string, error) {
 		logger.Warningf("service ip not found for mon %s. this better be a test", mon.Name)
 		return "", nil
 	}
+
 	logger.Infof("mon %s running at %s:%d", mon.Name, s.Spec.ClusterIP, mon.Port)
 	return s.Spec.ClusterIP, nil
 }
 
-func (c *Cluster) startPods(mons []*monConfig) error {
+func (c *Cluster) assignPods(mons []*monConfig) error {
 	// schedule the mons on different nodes if we have enough nodes to be unique
 	availableNodes, err := c.getAvailableMonNodes()
 	if err != nil {
 		return fmt.Errorf("failed to get available nodes for mons. %+v", err)
 	}
 
+	chosenNodes := map[string]*monConfig{}
 	nodeIndex := 0
 	for _, m := range mons {
-		// pick one of the available nodes where the mon will be assigned
-		node := availableNodes[nodeIndex%len(availableNodes)]
+		if _, ok := c.assignment[m.Name]; ok {
+			logger.Debugf("mon %s already assigned to a node, no need to assign", m.Name)
+			continue
+		}
+
+		// when the available nodes are >= than the mon size we just begin taking
+		// "node0" through nodeX (X = c.Size-1)
+		if len(availableNodes) >= c.Size {
+			// pick one of the available nodes where the mon will be assigned
+			c.assignment[m.Name] = availableNodes[nodeIndex]
+		} else {
+			// pick one of the available nodes where the mon will be assigned
+			node := availableNodes[nodeIndex%len(availableNodes)]
+			if _, ok := chosenNodes[node.Name]; !ok {
+				// node not used yet, update assignment and set that the node was chosen
+				chosenNodes[node.Name] = m
+				c.assignment[m.Name] = node
+				continue
+			}
+			// when the node was already chosen increase port by 1 and set
+			// assignment and that the node was chosen
+			m.Port = chosenNodes[node.Name].Port + 1
+			chosenNodes[node.Name] = m
```

Comment:
`chosenNodes` only needs to keep track of the last monitor assigned to get the next port number, right?

Reply:
@travisn Right, `chosendNodes` keeps track of the last monitor per node to get the next port number.